### PR TITLE
Implement OG protocol

### PR DESCRIPTION
**Pull request by anoited007**
Move header from body and restructure layout accordingly.

### CODE REVIEW

**Comment by anoited007** [comment]
@majordwarf I have done initial work on this #25 and want you to see if this meets what you are looking for before implementing in blog and modules. Also, if this is what you expect, kindly cross-check the OG values if they are valid.

File: src/templates/blog.jsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import SearchForm from "../components/BlogSearchForm/BlogSearchForm";
 import SearchResults from "../components/SearchResult/SearchResult";
 import config from "../../data/SiteConfig";
 import blogList from "../generated/blog-result.json";
+import favicon from "../../static/logos/logo.png";
```
Comment (majordwarf) [inline src/templates/blog.jsx]: No need of importing the favicon when you aren't gonna use it.

File: src/templates/post.jsx [hunk 2]
```diff
@@ -30,15 +30,16 @@ export default class PostTemplate extends React.Component {
           <div>
             <div className={"title"}>
               <h1>{post.title}</h1>
-              <h6>Posted by {post.author} on <span>{post.ddate}</span></h6>
+              <h6> Posted by {post.author} on <span>{post.ddate}</span></h6>
```
Comment (majordwarf) [inline src/templates/post.jsx]: Please remove the extra space before starting of the line.

File: src/layout/index.jsx [hunk 3]
```diff
@@ -9,15 +9,23 @@ import "bootstrap/dist/css/bootstrap.min.css";
...
 export default ({ children, location }) => (
-  <body>
+  <React.Fragment>
     <Helmet>
       <meta name="description" content={config.siteDescription} />
       <link rel="icon shortcut" href={favicon} type="image/png" />
+      <meta property="og:title" content={config.siteTitle} />
+      <meta property="og:type" content="website" />
+      <meta property="og:url" content={config.siteUrl} />
+      <meta property="og:image" content={favicon} />
```
Comment (majordwarf) [inline src/layout/index.jsx]: Favicon? The image property is used to generate the thumbnail that appears on the shared link's preview fetched by the webcrawlers. So, either use some image from our current website "terasology.org" or keep it blank as of now.

File: src/layout/index.jsx [hunk 4]
```diff
@@ -9,15 +9,23 @@ import "bootstrap/dist/css/bootstrap.min.css";
...
 
+
 // eslint-disable-next-line no-unused-vars
 export default ({ children, location }) => (
-  <body>
+  <React.Fragment>
     <Helmet>
       <meta name="description" content={config.siteDescription} />
       <link rel="icon shortcut" href={favicon} type="image/png" />
+      <meta property="og:title" content={config.siteTitle} />
```
Comment (majordwarf) [inline src/layout/index.jsx:19]: Using siteTitle property won't be a good choice here incase someone share a link to a some different page, that would only show the website title. If someone shares the link to about page it should display something like `About - Terasology`

You can have a look [here](https://neilpatel.com/wp-content/uploads/2014/03/Facebook-2014-03-03.jpg).

**Review by majordwarf — COMMENTED** [review]
Great going on the React Fragment! But still all the pages like post pages, module pages will show the same data as index page. That makes one of the main topic of issue yet unsolved, would be great to see how you approach it.

**Comment by anoited007** [comment]
@majordwarf Thank you for your feedback. I was hoping to get notified of your comments but unfortunately, there was no notification from GitHub on my part. I will look into your comments and work on them.

File: src/layout/index.jsx (hunk 4 above)
Comment (anoited007) [reply src/layout/index.jsx:19]: What should I use instead of siteTitle?

File: src/layout/index.jsx (hunk 3 above)
Comment (anoited007) [reply src/layout/index.jsx]: Alright. I think I will leave it blank for now or I can use siteLogo property if you want.

**Comment by anoited007** [comment]
I have made a few changes. I will update the og:title when we find a suitable property to use.

**Comment by majordwarf** [comment]
@anoited007 You're using the site name as the OG protocol's title for the post and blog page. We want them to be same as the blog name or the post name.

**Comment by anoited007** [comment]
@majordwarf  I have modified the OG title but the blog is not being concatenated to the URL. Can you have a look at and provide any help? I tried using '+' and other methods but no success. Last resort is probably saving it in a variable and using it as a literal.

**Pull request closed without merging**